File: temp/validate-slides.js
const fs = require('fs');
const path = require('path');

const slidesDir = 'D:/新课开发/自然科学/20.伪科学鉴别/授课PPT/slides';
const files = [42,47,83,84,85,86,88,89,90,91,92,93,94,95,96,97,99,102,103,104,105,106,107,109,110,111,113,114,115,116,117,118,119,123,128,131,135,140,142];

let ok = 0, fail = 0;
files.forEach(n => {
  const num = String(n).padStart(2,'0');
  const filePath = path.join(slidesDir, 'slide-' + num + '.js');
  try {
    new Function(fs.readFileSync(filePath, 'utf8'));
    ok++;
  } catch(e) {
    console.log(num + ': ' + e.message);
    fail++;
  }
});
console.log('OK: ' + ok + ', Failed: ' + fail);

File: temp/validate_slides.py
#!/usr/bin/env python3
"""Use Node.js vm to safely evaluate each slide and report errors."""
import subprocess, os, sys

SLIDES_DIR = r'D:\新课开发\内训师和表达\系列进阶课\12-内训项目运营：选师、建课、认证、应用与激励机制设计\授课PPT\slides'

# Slides to check
ALL_SLIDES = [f'slide-{i:02d}.js' for i in range(1, 131)]

for fname in ALL_SLIDES:
    fpath = os.path.join(SLIDES_DIR, fname)
    if not os.path.exists(fpath):
        continue

    # Use node to parse and report syntax errors
    code = f'''
const fs = require('fs');
const vm = require('vm');
try {{
  const src = fs.readFileSync('{fpath}', 'utf8');
  new vm.Script(src, {{ filename: '{fname}' }});
  console.log('OK');
}} catch(e) {{
  console.log('ERROR: ' + e.message);
}}
'''
    result = subprocess.run(
        ['node', '-e', code],
        capture_output=True, text=True,
        errors='replace'
    )
    if 'ERROR' in result.stdout:
        print(f'{fname}: {result.stdout.strip()}')
    elif 'OK' in result.stdout:
        pass  # silently OK
